src/sysupdate/sysupdate.c

```diff
@@ -1412,6 +1443,9 @@ static int run(int argc, char *argv[]) {
         if (r <= 0)
                 return r;
 
+        /* SIGCHLD signal must be blocked for sd_event_add_child to work */
+        assert_se(sigprocmask_many(SIG_BLOCK, NULL, SIGCHLD) >= 0);
```

Comment:
if you block a single signal only you don't really need sigprocmask_many()

Reply:
I've used BLOCK_SIGNALS() instead now, which I guess is generally also used to block more signals, but seems nicer to use than sigprocmask() by itself.

Reply:
> I've used BLOCK_SIGNALS() instead now, which I guess is generally also used to block more signals, but seems nicer to use than sigprocmask() by itself.

Please just use `sigprocmask`. `BLOCK_SIGNALS` is generally useful for blocking the signal for the surrounding block, but it be somewhat confusing here because you want to block this for the whole runtime of sysupdate.